JavaScript/sort/BucketSort.js

```diff
@@ -0,0 +1,47 @@
+function bucketSort(arr, bucketSize) {
```

Comment:
Add a beginner pitch of the algorithm along with space and time complexity in commented form.